### PR TITLE
Generate endpoint secret value

### PR DESCRIPTION
**Pull request by kmchicoine**
Generates a secret value that will be encrypted with a leader's public encryption key (once encryption keys are added to nodes) and attached to a commitment

### CODE REVIEW

File: web_server/src/lib.rs [hunk 1]
```diff
@@ -303,6 +375,7 @@ mod test {
 
     use super::*;
     use async_compatibility_layer::art::async_spawn;
+    //use hotshot_types::traits::signature_key::EncodedPublicKey;
```
Comment (DieracDelta) [inline web_server/src/lib.rs:378]: nit: Can this be deleted?

**Review by DieracDelta — DISMISSED** [review]
LGTM!

**Review by elliedavidson — DISMISSED** [review]

File: web_server/src/lib.rs [hunk 2]
```diff
@@ -277,6 +324,30 @@ where
...
+        async move {
+            let view_number: u64 = req.integer_param("view_number")?;
+            let secret: &str = req.string_param("secret")?;
+            //if secret is correct and view_number->proposal is empty, proposal is valid
+            if let Some(proposal) = state.proposals().get(&view_number) {
+                if proposal.0 == secret && proposal.1.is_empty() {
+                    let proposal = req.body_bytes();
+                    state.post_secret_proposal(view_number, proposal)
+                } else {
+                    Err(ServerError {
```
Comment (nyospe) [inline web_server/src/lib.rs:338]: Maybe split these conditions for distinct error messages.

File: web_server/src/lib.rs [hunk 3]
```diff
@@ -81,21 +83,22 @@ impl<KEY: SignatureKey + 'static> WebServerState<KEY> {
 
 /// Trait defining methods needed for the `WebServerState`
 pub trait WebServerDataSource<KEY> {
-    fn get_proposals(&self, view_number: u64) -> Result<Option<Vec<Vec<u8>>>, Error>;
+    fn get_proposals(&self, view_number: u64) -> Result<Option<Vec<u8>>, Error>;
```
Comment (nyospe) [inline web_server/src/lib.rs:86]: Shouldn't this be `get_proposal`?

**Review by nyospe — DISMISSED** [review]
A couple of minor nits, but nothing critical.

File: web_server/api.toml [hunk 4]
```diff
@@ -12,13 +12,13 @@ Return the proposal for a given view number
...
-":view_number" = "Integer"
-DOC = """
-Post the proposal for a given view_number
-"""
+#[route.postproposal]
+#PATH = ["proposal/:view_number"]
+#METHOD = "POST"
+#":view_number" = "Integer"
+#DOC = """
+#Post the proposal for a given view_number
```
[the remark this answers is not in the document]
Comment (kmchicoine) [reply web_server/api.toml:20]: Spent way too long chasing a CI bug before remembering that I had this commented out. It should definitely still be there

**Review by nyospe — APPROVED** [review]

**Pull request merged**